src/ring_theory/tensor_product.lean

```diff
@@ -296,12 +298,11 @@ We now build the structure maps for the symmetric monoidal category of `R`-algeb
 section monoidal
 
 section
-variables {R : Type u} {rR : comm_ring R}
-variables {A : Type v₁} {rA : ring A} {aA : algebra R A}
-variables {B : Type v₂} {rB : ring B} {aB : algebra R B}
-variables {C : Type v₃} {rC : ring C} {aC : algebra R C}
-variables {D : Type v₄} {rD : ring D} {aD : algebra R D}
-include aA aB aC
+variables {R : Type u} [comm_ring R]
+variables {A : Type v₁} [ring A] [algebra R A]
+variables {B : Type v₂} [ring B] [algebra R B]
+variables {C : Type v₃} [ring C] [algebra R C]
+variables {D : Type v₄} [ring D] [algebra R D]
```

Comment:
I'd previously been trying to gather these by unification rather than typeclass search, in the hope that would speed up the slow elaboration seen in `ring_theory/polynomial_algebra.lean`, but I tried out both and it seems to make no difference, so I'm reverting this back to the more straightforward setup.